lib/Sema/CSDiag.cpp

```diff
@@ -2316,10 +2303,13 @@ bool FailureDiagnosis::diagnoseContextualConversionError(
 
   // When complaining about conversion to a protocol type, complain about
   // conformance instead of "conversion".
-  if (contextualType->is<ProtocolType>() ||
-      contextualType->is<ProtocolCompositionType>())
-    diagID = diagIDProtocol;
-  
+  if (contextualType->isExistentialType()) {
```

Comment:
Should we do the same for opaque result types and/or AnyHashable? They have the same kinds of conformance-related issues.

Reply:
Yeah, those would already be diagnose by new diagnostic framework, I have a couple of examples added to the suite in my previous PR :)